src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -632,9 +634,15 @@ private Task ProcessSettingsFrameAsync()
             {
                 // int.MaxValue is the largest allowed windows size.
                 var previousInitialWindowSize = (int)_clientSettings.InitialWindowSize;
+                var previousMaxFrameSize = _clientSettings.MaxFrameSize;
 
                 _clientSettings.Update(_incomingFrame.GetSettings());
 
+                if (_clientSettings.MaxFrameSize != previousMaxFrameSize)
+                {
+                    _frameWriter.UpdateMaxFrameSize(_clientSettings.MaxFrameSize);
```

Comment:
I moved the ack to the start since there's a comment saying it should be done before updating the window. But I'm curious since I don't see why the current implementation has any downsides?